src/ShallowWrapper.js

```diff
@@ -1023,14 +1024,15 @@ class ShallowWrapper {
    */
   dive(options) {
     const name = 'dive';
+    const originalOptions = this.options;
     return this.single(name, (n) => {
       if (isDOMComponentElement(n)) {
         throw new TypeError(`ShallowWrapper::${name}() can not be called on DOM components`);
       }
       if (!isCustomComponentElement(n)) {
         throw new TypeError(`ShallowWrapper::${name}() can only be called on components`);
       }
-      return new ShallowWrapper(n, null, options);
+      return new ShallowWrapper(n, null, merge({}, originalOptions, options));
```

Comment:
I think this can use `this.options` inline - the arrow function will ensure `this` is correct.